### PR TITLE
Support WeChat as notifier

### PR DESCRIPTION
**Pull request by loganhz**
Now we don't support WeChat as a notifier. So user can not send the alert to wechat. We need to support it

https://github.com/rancher/rancher/issues/17121

Related PR
https://github.com/rancher/rancher/pull/17194

### CODE REVIEW

File: apis/management.cattle.io/v3/alerting_types.go [hunk 1]
```diff
@@ -300,6 +302,14 @@ type WebhookConfig struct {
 	URL string `json:"url,omitempty" norman:"required"`
 }
 
+type WechatConfig struct {
+	DefaultRecipient string `json:"defaultRecipient,omitempty" norman:"required"`
+	Secret           string `json:"secret,omitempty"`
```
Comment (aiwantaozi) [inline apis/management.cattle.io/v3/alerting_types.go]: It seems that API secret is required if you want to use wechat.

Comment (loganhz) [reply apis/management.cattle.io/v3/alerting_types.go]: Done

**Comment by aiwantaozi** [comment]
LGTM

File: apis/management.cattle.io/v3/alerting_types.go [hunk 2]
```diff
@@ -300,6 +302,14 @@ type WebhookConfig struct {
 	URL string `json:"url,omitempty" norman:"required"`
 }
 
+type WechatConfig struct {
+	DefaultRecipient string `json:"defaultRecipient,omitempty" norman:"required"`
+	Secret           string `json:"secret,omitempty" norman:"required"`
```
Comment (cjellick) [inline apis/management.cattle.io/v3/alerting_types.go]: Shouldnt the type be "password" for this field?

Comment (loganhz) [reply apis/management.cattle.io/v3/alerting_types.go]: Yes, it should be. Updated. Thanks!

**Comment by loganhz** [comment]
@cjellick Updated. Could you look at again, please?

**Comment by cjellick** [comment]
LGTM

**Pull request merged**